lib/cisco_node_utils/interface_port_channel.rb

```diff
@@ -0,0 +1,161 @@
+# December 2015, Sai Chintalapudi
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+require_relative 'interface'
+
+# Add some interface-specific constants to the Cisco namespace
+module Cisco
+  # InterfacePortChannel - node utility class for port channel config management
+  class InterfacePortChannel < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name.length > 0
+      @name = name.downcase
+      fail ArgumentError unless name.start_with?('port-channel')
+
+      create if instantiate
+    end
+
+    def self.interfaces
+      hash = {}
+      intf_list = config_get('interface', 'all_interfaces')
+      return hash if intf_list.nil?
+
+      intf_list.each do |id|
+        id = id.downcase
+        next unless id.start_with?('port-channel')
+        hash[id] = InterfacePortChannel.new(id, false)
+      end
+      hash
+    end
+
+    def create
+      config_set('interface_port_channel', 'create', @name)
+    end
+
+    def destroy
+      config_set('interface_port_channel', 'destroy', @name)
+    end
+
+    ########################################################
+    #                      PROPERTIES                      #
+    ########################################################
+
+    def lacp_graceful_convergence
+      config_get('interface_port_channel', 'lacp_graceful_convergence', @name)
+    end
+
+    def lacp_graceful_convergence=(state)
+      fail TypeError unless state == true || state == false
+      no_cmd = (state ? '' : 'no')
+      config_set('interface_port_channel',
+                 'lacp_graceful_convergence', @name, no_cmd)
+    rescue Cisco::CliError => e
+      raise "[#{@name}] '#{e.command}' : #{e.clierror}"
+    end
+
+    def default_lacp_graceful_convergence
+      config_get_default('interface_port_channel', 'lacp_graceful_convergence')
+    end
+
+    def lacp_max_bundle
+      config_get('interface_port_channel', 'lacp_max_bundle', @name)
+    end
+
+    def lacp_max_bundle=(val)
+      config_set('interface_port_channel', 'lacp_max_bundle', @name, '', val)
```

Comment:
I would expect to call config set with a `state` value instead of hard-coding `''`. If it will always be `''` then the state value can be removed from the yaml, but I suspect you'll want to use `state` in order to turn off the property.